File: nowcoder/把二叉树打印成多行.py
# encoding: utf-8

'''

@author: liangchi

@contact: bnu_llc@163.com

@software: pycharm

@file: 把二叉树打印成多行.py

@time: 2018/6/9 下午9:54

'''


class Solution:
    # 返回二维列表[[1,2],[4,5]]
    def Print(self, pRoot):
        # write code here
        Nodelist = []
        res = []

        if pRoot is None:
            return []
        Nodelist.append(pRoot)

        while Nodelist:
            templist = []
            n = len(Nodelist)
            for i in range(n):
                node = Nodelist.pop(0)
                templist.append(node.val)

                if node.left:
                    Nodelist.append(node.left)
                if node.right:
                    Nodelist.append(node.right)

            res.append(templist)
        return res

if __name__ == '__main__':
    pass

File: sword_c++/把二叉树打印成多行.cpp
//
// Created by 李先生 on 2018/9/4.
//

#include <iostream>
#include <queue>
using namespace std;

struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};

class Solution {
public:
    vector<vector<int> > Print(TreeNode* pRoot) {
        vector< vector<int>> res;
        if (pRoot == NULL)
            return res;

        queue<TreeNode *>  tqueue;

        tqueue.push(pRoot);

        while(!tqueue.empty())
        {
            vector<int> temp;
            int number = tqueue.size();
            while (number)
            {
                TreeNode * node = tqueue.front();
                tqueue.pop();
                temp.push_back(node->val) ;

                if (node->left)
                    tqueue.push(node->left);
                if(node->right)
                    tqueue.push(node->right);
                number --;
            }
            res.push_back(temp);
        }
        return res;
    }

};

int main()
{

}